dell/src/main/java/org/apache/iceberg/dell/ecs/EcsAppendOutputStream.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.dell.ecs;
+
+import com.emc.object.s3.S3Client;
+import com.emc.object.s3.request.PutObjectRequest;
+import java.io.ByteArrayInputStream;
+import java.nio.ByteBuffer;
+import org.apache.iceberg.io.PositionOutputStream;
+
+/**
+ * Use ECS append API to write data.
+ */
+class EcsAppendOutputStream extends PositionOutputStream {
+
+  private final S3Client client;
+
+  private final EcsURI key;
+
+  /**
+   * Local bytes cache that avoid too many requests
+   * <p>
+   * Use {@link ByteBuffer} to maintain offset.
+   */
+  private final ByteBuffer localCache;
+
+  /**
+   * A marker for data file to put first part instead of append first part.
+   */
+  private boolean firstPart = true;
+
+  /**
+   * Pos for {@link PositionOutputStream}
+   */
+  private long pos;
+
+  private EcsAppendOutputStream(S3Client client, EcsURI key, byte[] localCache) {
```

Comment:
Nit: I'd prefer to change the name `key` to `uri`.

Reply:
Changed!